code/implementation/vm/src/main/java/io/cattle/platform/vm/process/VirtualMachinePreCreate.java

```diff
@@ -98,10 +102,20 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         Object objectDisks = DataAccessor.field(instance, InstanceConstants.FIELD_DISKS, Object.class);
         if (objectDisks instanceof List<?>) {
             String namePrefix = instance.getName();
+            Long svcIndexId = DataAccessor.fieldLong(instance, InstanceConstants.FIELD_SERVICE_INSTANCE_SERVICE_INDEX_ID);
+
+            String uuidPart = null;
+            if (svcIndexId != null) {
+                Service svc = serviceDao.getServiceByServiceIndexId(svcIndexId);
+                uuidPart = svc.getUuid().substring(0, 7);
```

Comment:
Can we add the service index to that volume name.  So if I scale the service to more than 1 I don't end up trying to use the same volume.